I see some issues like this:

```
/home/travis/build/Dead2/zlib-ng/memcopy.h:24:5: warning: ISO C90 forbids mixed declarations and code [-Wpedantic]
     uint32_t res = chunk.a;
     ^
/home/travis/build/Dead2/zlib-ng/memcopy.h: In function ‘set_bytes’:
/home/travis/build/Dead2/zlib-ng/memcopy.h:193:9: warning: ISO C90 forbids mixed declarations and code [-Wpedantic]
         unsigned char c = *from;
         ^
/home/travis/build/Dead2/zlib-ng/memcopy.h: In function ‘chunk_memcpy’:
/home/travis/build/Dead2/zlib-ng/memcopy.h:230:5: warning: ISO C90 forbids mixed declarations and code [-Wpedantic]
     unsigned rem = len % sz;
     ^
/home/travis/build/Dead2/zlib-ng/memcopy.h:235:5: warning: ISO C90 forbids mixed declarations and code [-Wpedantic]
     unsigned by8 = len % sz;
     ^
/home/travis/build/Dead2/zlib-ng/memcopy.h: In function ‘byte_memset’:
/home/travis/build/Dead2/zlib-ng/memcopy.h:290:5: warning: ISO C90 forbids mixed declarations and code [-Wpedantic]
     unsigned char *from = out - 1;
     ^
/home/travis/build/Dead2/zlib-ng/memcopy.h:295:5: warning: ISO C90 forbids mixed declarations and code [-Wpedantic]
     unsigned rem = len % sz;
     ^
/home/travis/build/Dead2/zlib-ng/memcopy.h:300:5: warning: ISO C90 forbids mixed declarations and code [-Wpedantic]
     unsigned by8 = len % 8;
     ^
/home/travis/build/Dead2/zlib-ng/memcopy.h: In function ‘chunk_memset’:
/home/travis/build/Dead2/zlib-ng/memcopy.h:359:5: warning: ISO C90 forbids mixed declarations and code [-Wpedantic]
     unsigned sz = sizeof(uint64_t);
```

What is your preference? I can move the variables before `memcpy`.

--- a/arch/x86/insert_string_sse.c
+++ b/arch/x86/insert_string_sse.c
@@ -23,7 +23,7 @@ ZLIB_INTERNAL Pos insert_string_sse(deflate_state *const s, const Pos str, unsig
 
     for (idx = 0; idx < count; idx++) {
         ip = (unsigned *)&s->window[str+idx];
-        memcpy(&val, ip, sizeof(val)); // val = *ip;
+        memcpy(&val, ip, sizeof(val));
         h = 0;
 
         if (s->level >= TRIGGER_LEVEL)

--- a/match.c
+++ b/match.c
@@ -363,6 +363,7 @@ ZLIB_INTERNAL unsigned longest_match(deflate_state *const s, IPos cur_match) {
     register unsigned char *strend = s->window + s->strstart + MAX_MATCH;
 
     uint16_t scan_start, scan_end;
+
     memcpy(&scan_start, scan, sizeof(scan_start));
     memcpy(&scan_end, scan+best_len-1, sizeof(scan_end));
 

--- a/match.c
+++ b/match.c
@@ -428,10 +428,12 @@ ZLIB_INTERNAL unsigned longest_match(deflate_state *const s, IPos cur_match) {
         scan += 2, match+=2;
         Assert(*scan == *match, "match[2]?");
         do {
-            unsigned long sv, mv;
+            unsigned long sv, mv, xor;
+            
             memcpy(&sv, scan, sizeof(sv)); // = *(unsigned long*)(void*)scan;
             memcpy(&mv, match, sizeof(mv)); // = *(unsigned long*)(void*)match;
-            unsigned long xor = sv ^ mv;
+            xor = sv ^ mv;
+            
             if (xor) {
                 int match_byte = __builtin_ctzl(xor) / 8;
                 scan += match_byte;

--- a/match.c
+++ b/match.c
@@ -361,11 +361,11 @@ ZLIB_INTERNAL unsigned longest_match(deflate_state *const s, IPos cur_match) {
     unsigned int wmask = s->w_mask;
 
     register unsigned char *strend = s->window + s->strstart + MAX_MATCH;
-    
+
     uint16_t scan_start, scan_end;
-    memcpy(&scan_start, scan, sizeof(scan_start)); // = *(uint16_t*)scan;
-    memcpy(&scan_end, scan+best_len-1, sizeof(scan_end)); // = *(uint16_t*)(scan+best_len-1);
-    
+    memcpy(&scan_start, scan, sizeof(scan_start));
+    memcpy(&scan_end, scan+best_len-1, sizeof(scan_end));
+
     /* The code is optimized for HASH_BITS >= 8 and MAX_MATCH-2 multiple of 16.
      * It is easy to get rid of this optimization if necessary.
      */
@@ -429,11 +429,12 @@ ZLIB_INTERNAL unsigned longest_match(deflate_state *const s, IPos cur_match) {
         Assert(*scan == *match, "match[2]?");
         do {
             unsigned long sv, mv, xor;
-            
-            memcpy(&sv, scan, sizeof(sv)); // = *(unsigned long*)(void*)scan;
-            memcpy(&mv, match, sizeof(mv)); // = *(unsigned long*)(void*)match;
+
+            memcpy(&sv, scan, sizeof(sv));
+            memcpy(&mv, match, sizeof(mv));
+
             xor = sv ^ mv;
-            
+
             if (xor) {
                 int match_byte = __builtin_ctzl(xor) / 8;
                 scan += match_byte;

